module/VuFind/src/VuFind/RecordTab/AbstractBase.php

```diff
@@ -110,6 +110,16 @@ public function supportsAjax()
         return true;
     }
 
+    /**
+     * What CSS and JS dependencies does this tab have?
+     *
+     * @return array
+     */
+    public function getDependencies()
```

Comment:
This approach kind of blurs the lines between front end and back end code, and also makes it impossible to use different carousel implementations across different themes (which seems an unlikely requirement, but something that would be nice to be able to do in theory).

Does the back-end code really need to know about the front-end dependencies? Might a better approach be to simply load this content in the SimilarItemsCarousel.phtml file instead?

Also, on a related note, I feel like this is going to cause problems for AJAX-loaded tabs unless we are careful about our approach. I'm not seeing anything obviously broken in my tests, but it would probably be a good idea to test that closely when time permits.

Reply:
You're right. This was probably an over-reaction to including a relatively larger library. I'll weed this out.